x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/action/PutDataFrameTransformAction.java

```diff
@@ -58,6 +58,11 @@ public Request(DataFrameTransformConfig config) {
         public Request(StreamInput in) throws IOException {
             super(in);
             this.config = new DataFrameTransformConfig(in);
+            if (in.getVersion().onOrAfter(Version.CURRENT)) {
```

Comment:
as for check in BWC code like this: you can alternatively set the version, disable BWC in `build.gradle`, do the backports and then re-enable BWC. You need the same number of commits in the end, but using the "`Version.CURRENT` approach" still broke CI, disabling BWC seems to be the better approach now.

Reply:
@hendrikmuhs I usually mute the tests before I backport in a separate PR. This way BWC can actually run against my changes and I can find any issues earlier rather than later.